src/index.rs

```diff
@@ -516,9 +563,41 @@ impl ResourceIndex {
         self.path2id.insert(path, entry);
     }
 
+    /// Removes the given resource ID from the index and returns an update
+    /// containing the deleted entries
+    pub fn forget_id(&mut self, old_id: ResourceId) -> Result<IndexUpdate> {
+        log::debug!("Forgetting a single entry in the index");
+
+        // Collect all paths associated with the old ID
+        let mut old_paths = Vec::new();
+        for (path, entry) in &self.path2id {
+            if entry.id == old_id {
+                old_paths.push(path.clone());
+            }
+        }
+
+        // Remove entries from path2id and id2path
+        for path in &old_paths {
+            self.path2id.remove(path);
+        }
+        self.id2path.remove(&old_id);
+
+        let mut deleted = HashSet::new();
+        deleted.insert(old_id);
+
+        Ok(IndexUpdate {
+            added: HashMap::new(),
+            deleted,
+        })
+    }
+
+    /// Removes an entry with the specified path and updates the collision
+    /// information accordingly
+    ///
+    /// Returns an update containing the deleted entries
     fn forget_path(
         &mut self,
-        path: &CanonicalPath,
+        path: &Path,
         old_id: ResourceId,
     ) -> Result<IndexUpdate> {
```

Comment:
1. I would return just `deleted` collection, because `added` is always empty here.
2. Maybe name it `forget_entry` since path and id have same importance?

Reply:
> I would return just `deleted` collection, because `added` is always empty here.
  
That's why I suggest finalizing the API schema in #91 before implementing changes in a separate PR. #72 modifies `IndexUpdate`, adds its `Default` implementation, and integrates it into multiple API methods. 

Trying to merge #72 changes into this PR while making further modifications along the way could become complicated.